lib/krane/kubernetes_resource/deployment.rb

```diff
@@ -91,6 +91,10 @@ def deploy_timed_out?
       return false if deploy_failed?
       return super if timeout_override
 
+      if progress_condition.present?
+        StatsD.client.increment('kubectl.fail_fast', 1, tags: statsd_tags)
+      end
+
```

Comment:
This will be evaluated on every sync loop, which happens X times per deployment. is that what you want?
Also, when we hit this code, we're not necessarily failing fast, no? We still need to observe that on line 99

Reply:
🤦 Sorry I realize you are capturing the specifics in the actual statsd_tag call